src/algokit/core/deploy.py

```diff
@@ -0,0 +1,157 @@
+# 1. User can call algokit deploy to different networks
+# 2. By default algokit cli contains configs for testnet and mainnet
+# 3. User can overwrite them by creating a config file in the project root
+
+
+import configparser
+import logging
+import os
+import sys
+from pathlib import Path
+
+import click
+
+from algokit.core import proc
+from algokit.core.bootstrap import ALGOKIT_CONFIG
+
+if sys.version_info >= (3, 11):
+    import tomllib
+else:
+    import tomli as tomllib
+
+logger = logging.getLogger(__name__)
+
+
+def _generate_config(  # noqa: PLR0913
+    algod_server: str,
+    indexer_server: str,
+    algod_token: str = "",
+    algod_port: str = "",
+    indexer_token: str = "",
+    indexer_port: str = "",
+) -> dict[str, str]:
+    return {
+        "ALGOD_TOKEN": algod_token,
+        "ALGOD_SERVER": algod_server,
+        "ALGOD_PORT": algod_port,
+        "INDEXER_TOKEN": indexer_token,
+        "INDEXER_SERVER": indexer_server,
+        "INDEXER_PORT": indexer_port,
+    }
+
+
+REQUIRED_KEYS = [
+    "ALGOD_TOKEN",
+    "ALGOD_SERVER",
+    "ALGOD_PORT",
+    "INDEXER_TOKEN",
+    "INDEXER_SERVER",
+    "INDEXER_PORT",
+]
+
+DEPLOY_CONFIGS = {
+    "localnet": _generate_config(
+        algod_token="aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa",
+        algod_server="http://localhost",
+        algod_port="4001",
+        indexer_token="aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa",
+        indexer_server="http://localhost",
+        indexer_port="8980",
+    ),
+    "testnet": _generate_config(
+        algod_server="https://testnet-api.algonode.cloud",
+        indexer_server="https://testnet-idx.algonode.cloud",
+    ),
+    "mainnet": _generate_config(
+        algod_server="https://mainnet-api.algonode.cloud",
+        indexer_server="https://mainnet-idx.algonode.cloud",
+    ),
+    "betanet": _generate_config(
+        algod_server="https://betanet-api.algonode.cloud",
+        indexer_server="https://betanet-idx.algonode.cloud",
+    ),
+}
+
+
+def load_deploy_config(network: str, project_dir: Path) -> dict:
+    """
+    Parse the .env file if it exists and load configurations.
+    :param network: Network name.
+    :param project_dir: Project directory path.
+    :return: A dictionary containing the configuration.
+    """
+
+    env_file_path = project_dir / ".env"
+    if env_file_path.exists():
+        config = configparser.ConfigParser()
```

Comment:
we could use https://github.com/theskumar/python-dotenv#load-configuration-without-altering-the-environment instead here?

Reply:
added new dep to poetry - now relies on dotent package